02.ComputeShader/main.cpp

```diff
@@ -443,30 +444,17 @@ class ComputeShaderSampleApp : public ApplicationBase
 		auto& cb = m_cmdbuf[m_resourceIx];
 		auto& commandPool = commandPools[CommonAPI::InitOutput::EQT_COMPUTE][m_resourceIx];
 		auto& fence = m_frameComplete[m_resourceIx];
-		if (fence)
-		{
-			logicalDevice->blockForFences(1u, &fence.get());
-			logicalDevice->resetFences(1u, &fence.get());
-			if (m_frameIx >= FRAMES_IN_FLIGHT) CommonAPI::dropRetiredSwapchainResources(m_qRetiredSwapchainResources, m_frameIx - FRAMES_IN_FLIGHT);
-		}
-		else
-			fence = logicalDevice->createFence(static_cast<video::IGPUFence::E_CREATE_FLAGS>(0));
 
-		// acquire image 
-		uint32_t imgnum = 0u;
-		std::unique_lock guard = acquireImage(&imgnum);
+		waitForFrame(FRAMES_IN_FLIGHT, fence);
+		logicalDevice->resetFences(1u, &fence.get());
```

Comment:
why not make the `waitForFrame` reset the fence after blocking? (fold it into itself)